service-base/src/test/java/org/eclipse/hono/service/management/device/AutoProvisioningEnabledDeviceBackendTest.java

```diff
@@ -80,11 +80,9 @@ public void testProvisionDeviceSucceeds(final VertxTestContext ctx) {
         final AutoProvisioningEnabledDeviceBackend underTest = mock(AutoProvisioningEnabledDeviceBackend.class);
         when(underTest.provisionDevice(anyString(), any(), any())).thenCallRealMethod();
 
-        doAnswer(invocation -> {
-            final Promise<OperationResult<Id>> promise = invocation.getArgument(4);
-            promise.complete(OperationResult.ok(201, Id.of(DEVICE_ID), Optional.empty(), Optional.empty()));
-            return null;
-        }).when(underTest).createDevice(any(), any(), any(), any(), any(Handler.class));
+        doAnswer(invocation -> Future
+                .succeededFuture(OperationResult.ok(201, Id.of(DEVICE_ID), Optional.empty(), Optional.empty()))
+        ).when(underTest).createDevice(any(), any(), any(), any());
```

Comment:
Using `when(underTest.createDevice(...)).thenReturn(Future.succeededFuture(...));` looks simpler here.

Same in 2 other places below.

Reply:
I though of the same and I tried to keep it as close to the original code, so its easier to review. I changed it now.